prow.go

```diff
@@ -834,13 +836,61 @@ func getClusterClient(m *jobManager, job *Job) (*BuildClusterClientConfig, error
 	return clusterClient, nil
 }
 
+func createAccessRBAC(namespace string, user string, client ctrlruntimeclient.Client) error {
+	roleBinding := &rbacapi.RoleBinding{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "cluster-bot-user-access",
+			Namespace: namespace,
+		},
+		Subjects: []rbacapi.Subject{{Kind: "User", Name: user}},
+		RoleRef: rbacapi.RoleRef{
+			Kind: "ClusterRole",
+			Name: "admin",
+		},
+	}
+	if err := client.Create(context.Background(), roleBinding); err != nil && !errors.IsAlreadyExists(err) {
+		return err
+	}
+	return nil
+}
+
+func (m *jobManager) setupAccessRBAC(job *Job, namespace string) {
+	err := wait.PollImmediate(5*time.Second, 5*time.Minute, func() (bool, error) {
+		clusterClient, err := getClusterClient(m, job)
+		if err != nil {
+			return false, err
+		}
+		if job.RequesterUserID != "" {
+			client, err := ctrlruntimeclient.New(clusterClient.CoreConfig, ctrlruntimeclient.Options{})
+			if err != nil {
+				return false, err
+			}
+			if err := createAccessRBAC(namespace, job.RequesterUserID, client); err != nil {
+				klog.Infof("could not create role binding for %s: %v", job.RequesterUserID, err)
+				// the namespace might not yet exist when this step is executed
+				// we want to retry if this step fails, hence the nil return
+				return false, nil
```

Comment:
maybe it should be split to two conditions (for checking existence of the namespace and creation of rbac)?
that's because rbac creation might be because of other reasons

Reply:
Generally, we want to retry on any type of failure.
The namespace one is noted since it occurs quite frequently (due to a race condition).